src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,120 @@
+import * as ld from "lodash";
+
+// First Group will be the variable value. Doesn't fetch the variable names
+const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+
+export class PrologResult {
+	private readonly rawResults: string[];
+	private readonly results: Map<string, Map<number, string|boolean>>;
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = new Map();
+	}
+
+	public getResults(): Map<string, Map<number, string|boolean>> {
+		if (this.results.size !== 0) { return this.results; }
+
+		const results = new Map();
+		let i: number;
+
+		this.getVariables().forEach((e: string) => {
+			results.set(e, new Map());
+			i = 0;
+
+			this.getResultsFor(e).forEach((e2: string) => {
+				results.get(e).set(i, e2);
+				i++;
+			});
+
+		});
+
+		results.set("Boolean", new Map());
+		i = 0;
+
+		this.getBooleans().forEach((e: boolean) => {
+			results.get("Boolean").set(i, e);
+			i++;
+		});
+
+		return results;
+	}
+
+	public getResultArray(): string[][] {
+		const results: string[][] = [];
+		this.rawResults.forEach((e: string) => {
+			results.push(this.rawResultsToInnerArray(e));
+		});
+
+		return results;
+	}
+
+	private getResultsFor(variable: string): string[] {
+		// Group 1 will be the content of the variable independent of the variable length.
+	 const pattern: any = new RegExp(variable + prologAnswerPattern, "g");
+	 return this.regexGroupToArray(this.rawResults.toString(), pattern, 1);
+	}
+
+	private rawResultsToInnerArray(source: string): string[] {
+
+	if (source.includes("true")) { return ["true"]; }
```

Comment:
Not sure what valid values for `source` are, but couldn't you also have something like `Gertrued` (misspelled name :D)? In that case I don't think you'd want to return `["true"]`?